services/travis/travis-php-version.service.js

```diff
@@ -1,29 +1,39 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const log = require('../../core/server/log')
+const Joi = require('joi')
+const { BaseJsonService } = require('..')
 const {
-  minorVersion: phpMinorVersion,
-  versionReduction: phpVersionReduction,
+  minorVersion,
+  versionReduction,
   getPhpReleases,
 } = require('../php-version')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class TravisPhpVersion extends LegacyService {
+const optionalNumberOrString = Joi.alternatives(Joi.string(), Joi.number())
+const schema = Joi.object({
+  branch: Joi.object({
+    config: Joi.object({
+      php: Joi.array().items(optionalNumberOrString),
+      matrix: Joi.object({
+        include: Joi.array().items(Joi.object({ php: optionalNumberOrString })),
+      }),
+    }).required(),
+  }).required(),
+}).required()
+
+module.exports = class TravisPhpVersion extends BaseJsonService {
+  constructor(context, config) {
```

Comment:
May want to go ahead and move this down between `render` and `transform` given #3353